src/ResourceManager/IotHub/Commands.IotHub.Test/ScenarioTests/IotHubTests.ps1

```diff
@@ -0,0 +1,141 @@
+
+	# Get Connection string
+	$connectionstring = Get-AzureRmIotHubConnectionString -ResourceGroupName $ResourceGroupName -Name $IotHubName -KeyName iothubowner
+	Assert-True { $key.KeyName -eq "iothubowner" }
+
+	# Add consumer group
+	Add-AzureRmIotHubEventHubConsumerGroup -ResourceGroupName $ResourceGroupName -Name $IotHubName -EventHubEndpointName events -EventHubConsumerGroupName cg1
+
+	# Get consumer group
+	$eventubConsumerGroup = Get-AzureRmIotHubEventHubConsumerGroup -ResourceGroupName $ResourceGroupName -Name $IotHubName -EventHubEndpointName events
+	Assert-True { $eventubConsumerGroup.Count -eq 2 }
+
+	# Delete consumer group
+	Remove-AzureRmIotHubEventHubConsumerGroup -ResourceGroupName $ResourceGroupName -Name $IotHubName -EventHubEndpointName events -EventHubConsumerGroupName cg1
+
+	# Get consumer group
+	$eventubConsumerGroup = Get-AzureRmIotHubEventHubConsumerGroup -ResourceGroupName $ResourceGroupName -Name $IotHubName -EventHubEndpointName events
+	Assert-True { $eventubConsumerGroup.Count -eq 1 }
+
+	# Add Key
+	Add-AzureRmIotHubKey -ResourceGroupName $ResourceGroupName -Name $IotHubName -KeyName iothubowner1 -PrimaryKey 4GT/3sQXHYLDVOG5c8GQCpxIAw+OQtE5RxpdFC6O5Jk= -SecondaryKey 4GT/3sQXHYLDVOG5c8GQCpxIAw+OQtE5RxpdFC6O5Jk= -Rights RegistryRead
+
+	# Get Keys
+	$keys = Get-AzureRmIotHubKey -ResourceGroupName $ResourceGroupName -Name $IotHubName 
+	Assert-True { $keys.Count -eq 6 }
+
+	# Remove Key
+	Remove-AzureRmIotHubKey -ResourceGroupName $ResourceGroupName -Name $IotHubName -KeyName iothubowner1
+
+	# Get Keys
+	$keys = Get-AzureRmIotHubKey -ResourceGroupName $ResourceGroupName -Name $IotHubName 
+	Assert-True { $keys.Count -eq 5 }
+
+	# Set Sku
+	$iothub = Get-AzureRmIotHub -ResourceGroupName $ResourceGroupName -Name $IotHubName 
+	$iothubUpdated = Set-AzureRmIotHub -ResourceGroupName $ResourceGroupName -Name $IotHubName -SkuName S1 -Units 5
+	Assert-True { $iothubUpdated.Sku.Capacity -eq 5 }
+
+	# Event Hub Properties Update
+	$iothubUpdated = Set-AzureRmIotHub -ResourceGroupName $ResourceGroupName -Name $IotHubName -EventHubRetentionTimeInDays 5
+	Assert-True { $iothubUpdated.Properties.EventHubEndpoints.events.RetentionTimeInDays -eq 5 }
+
+	# Cloud To Device Properties Update
+	$cloudToDevice = $iothubUpdated.Properties.CloudToDevice
+	$cloudToDevice.MaxDeliveryCount = 25
+	$iotHubUpdated = Set-AzureRmIotHub -ResourceGroupName $ResourceGroupName -Name $IotHubName -CloudToDevice $cloudToDevice
+	Assert-True { $iothubUpdated.Properties.CloudToDevice.MaxDeliveryCount -eq 25 }
+
+	# Operations Monitoring properties update
+	$op= $iotHubUpdated.Properties.OperationsMonitoringProperties
+	$op.OperationMonitoringEvents["Connections"] = "Information"
+	$iotHubUpdated = Set-AzureRmIotHub -ResourceGroupName $ResourceGroupName -Name $IotHubName -OperationsMonitoringProperties $op
+	Assert-True { $iothubUpdated.Properties.OperationsMonitoringProperties.OperationMonitoringEvents["Connections"] -eq "Information" }
```

Comment:
Not as a part of the current PR. The test for this is a little involved as it requires a SAS token to be fetched on the fly. Will add the test later.